panel/io/ipywidget.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+from bokeh.document.events import MessageSentEvent
+from ipykernel.comm import CommManager
+from ipywidgets_bokeh.kernel import BokehKernel, SessionWebsocket, WebsocketStream
+
+
+class PanelSessionWebsocket(SessionWebsocket):
+
+    def __init__(self, *args, **kwargs):
+        self._document = kwargs.pop('document', None)
+        self._queue = []
+        super(PanelSessionWebsocket, self).__init__(*args, **kwargs)
+
+    def send(self, stream, msg_type, content=None, parent=None, ident=None, buffers=None, track=False, header=None, metadata=None):
+        msg = self.msg(msg_type, content=content, parent=parent, header=header, metadata=metadata)
+        msg['channel'] = stream.channel
+
+        doc = self._document
```

Comment:
I would add:
```py
@property
def document(self):
    return curdoc()
```
in `ipywidgets_bokeh` and override it here with `self._document`. Similar thing can be done with even trigger, and all this code duplication could be avoided.

Reply:
Right I was going to make a PR on ipywidgets_bokeh to reduce code duplication.